wordpress-mobile/WordPress-Android
# Issue/5681 discard changes action

### Fix
Add action to overflow menu to discard local changes and snackbar with undo action to revert discarded changes to post editor and preview to address https://github.com/wordpress-mobile/WordPress-Android/issues/5681.  See the screenshots and animation below for illustration.

![discard_changes](https://user-images.githubusercontent.com/3827611/42412465-04d89c6e-81ca-11e8-9c9a-985efee2a103.png)

<img src="https://user-images.githubusercontent.com/3827611/42406999-dece6e1c-8170-11e8-950d-30dbf4f4f146.gif" width="360">

### Test
1. Go to ***My Site*** tab.
2. Tap ***Blog Posts*** item under ***Publish*** section.
3. Tap post with local changes.
4. Notice local changes in post editor.
5. Tap overflow menu action.
6. Tap ***Discard Local Changes*** action.
7. Notice local changes are removed in post editor.
8. Notice ***Local changes discarded*** snackbar with ***Undo*** action.
9. Tap ***Undo*** action.
10. Notice local changes are added in post editor.
